src/sympc/protocol/aby3/aby3.py

```diff
@@ -296,3 +313,110 @@ def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
         arith_share = x1 ^ x2 ^ x3
 
         return arith_share
+
+    @staticmethod
+    def full_adder(
+        a: List[MPCTensor], b: List[MPCTensor], session: Session
+    ) -> List[MPCTensor]:
+        """Perform bit addition on MPCTensors using a full adder.
+
+        Args:
+            a (List[MPCTensor]): MPCTensor with shares of bit.
+            b (List[MPCTensor]): MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+
+        Returns:
+            result (List[MPCTensor]): Result of the operation.
+
+        TODO: Should modify ripple carry adder to parallel prefix adder,currently unused.
+        """
+        ring_size = session.ring_size
+        ring_bits = get_nr_bits(ring_size)
+        c = [0 for _ in range(ring_bits + 1)]  # carry bits of addition.
+        result: List[MPCTensor] = []
+        for idx in range(ring_bits):
+            s = a[idx] + b[idx] + c[idx]
+            c[idx + 1] = a[idx] * b[idx] + c[idx] * (a[idx] + b[idx])
+            result.append(s)
+        return result
+
+    @staticmethod
+    def bit_decomposition(x: MPCTensor, session: Session) -> List[MPCTensor]:
+        """Perform ABY3 bit decomposition for conversion of arithmetic share to binary share.
+
+        Args:
+            x (MPCTensor): Arithmetic shares of secret.
+            session (Session): session the share belongs to.
+
+        Returns:
+            bin_share (List[MPCTensor]): Returns binary shares of each bit of the secret.
+
+        TODO : Should be modified to use parallel prefix adder when multiprocessing
+        functionality is integrated,currently unused.
+        """
+        ring_size = session.ring_size
+        ring_bits = get_nr_bits(ring_size)
+        x1: List[MPCTensor] = []  # bit shares of shares
+        x2: List[MPCTensor] = []
+        x3: List[MPCTensor] = []
+
+        args = [[share, str(2), True] for share in x.share_ptrs]
```

Comment:
Q: Why not directly "2"?

Reply:
The ring_size (`2**64`) exceeded protobuf int,64 range. so I used string george.

Reply:
No no...I meant. Why not `"2"` and not `str(2)`

Reply:
Oh sorry I misinterpreted, sure I would change it